### ISSUE TITLE
Copying a page twice fails when Postgres search is enabled

### ISSUE THREAD

**Issue by gasman** [wagtail/wagtail#4120]
### Issue Summary

The postgres search backend adds a generic index_entries relation to indexed models, which causes copying to fail with `'GenericRelatedObjectManager' object is not iterable`. This is probably a manifestation of #3920, but the solution arrived at there, of introducing an `excluded_fields_for_copy` attribute on pages, isn't appropriate because the relation is coming from Wagtail itself rather than something defined by the user.

(Note: this may well affect 1.12/1.13 too, and should probably be backported if so.)

### Steps to Reproduce

1. Start from bakerydemo, configured to use Postgresql as database and search backend
2. Log in and go to the homepage explorer view
3. Copy the 'Contact Us' page (via More -> Copy, changing title/slug to Contact Us 2)
4. Copy the 'Contact Us' page again (changing title/slug to Contact Us 3)

Receive the error:

    Environment:

    Request Method: POST
    Request URL: http://localhost:8000/admin/pages/69/copy/

    Django Version: 1.11.8
    Python Version: 3.5.2
    Installed Applications:
    ['bakerydemo.base',
     'bakerydemo.blog',
     'bakerydemo.breads',
     'bakerydemo.locations',
     'bakerydemo.search',
     'wagtail.contrib.search_promotions',
     'wagtail.contrib.forms',
     'wagtail.contrib.redirects',
     'wagtail.embeds',
     'wagtail.sites',
     'wagtail.users',
     'wagtail.snippets',
     'wagtail.documents',
     'wagtail.images',
     'wagtail.search',
     'wagtail.admin',
     'wagtail.contrib.modeladmin',
     'wagtail.core',
     'rest_framework',
     'modelcluster',
     'taggit',
     'wagtailfontawesome',
     'django.contrib.admin',
     'django.contrib.auth',
     'django.contrib.contenttypes',
     'django.contrib.sessions',
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'wagtail.contrib.postgres_search']
    Installed Middleware:
    ['django.middleware.security.SecurityMiddleware',
     'django.contrib.sessions.middleware.SessionMiddleware',
     'django.middleware.common.CommonMiddleware',
     'django.middleware.csrf.CsrfViewMiddleware',
     'django.contrib.auth.middleware.AuthenticationMiddleware',
     'django.contrib.messages.middleware.MessageMiddleware',
     'django.middleware.clickjacking.XFrameOptionsMiddleware',
     'wagtail.core.middleware.SiteMiddleware',
     'wagtail.contrib.redirects.middleware.RedirectMiddleware']

    Traceback:

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/core/handlers/exception.py" in inner
      41.             response = get_response(request)

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
      187.                 response = self.process_exception_by_middleware(e, request)

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/core/handlers/base.py" in _get_response
      185.                 response = wrapped_callback(request, *callback_args, **callback_kwargs)

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/views/decorators/cache.py" in _cache_controlled
      43.             response = viewfunc(request, *args, **kw)

    File "/vagrant/wagtail/wagtail/admin/urls/__init__.py" in wrapper
      95.             return view_func(request, *args, **kwargs)

    File "/vagrant/wagtail/wagtail/admin/decorators.py" in decorated_view
      28.             return view_func(request, *args, **kwargs)

    File "/vagrant/wagtail/wagtail/admin/utils.py" in wrapped_view_func
      138.                 return view_func(request, *args, **kwargs)

    File "/vagrant/wagtail/wagtail/admin/views/pages.py" in copy
      820.                 user=request.user,

    File "/vagrant/wagtail/wagtail/core/models.py" in copy
      1060.         page_copy = self.specific_class(**specific_dict)

    File "/vagrant/wagtail/wagtail/contrib/forms/models.py" in __init__
      161.         super().__init__(*args, **kwargs)

    File "/vagrant/wagtail/wagtail/core/models.py" in __init__
      358.         super().__init__(*args, **kwargs)

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/modelcluster/models.py" in __init__
      177.             super(ClusterableModel, self).__init__(*args, **kwargs)

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/db/models/base.py" in __init__
      568.                             _setattr(self, prop, kwargs[prop])

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/db/models/fields/related_descriptors.py" in __set__
      537.         manager.set(value)

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/contrib/contenttypes/fields.py" in set
      620.             objs = tuple(objs)

    Exception Type: TypeError at /admin/pages/69/copy/
    Exception Value: 'GenericRelatedObjectManager' object is not iterable

Examining local vars of the item:

    File "/home/ubuntu/.virtualenvs/bakerydemo/lib/python3.5/site-packages/django/db/models/base.py" in __init__
      568.                             _setattr(self, prop, kwargs[prop])

shows that the failure is happening when assigning the `index_entries` property.

### Technical details

* Python version: 3.5.2
* Django version: 1.11.8
* Wagtail version: 2.0a0
* Browser version: Chrome 62

**Comment by BertrandBordage** [comment]
This is already [fixed in #4059](https://github.com/wagtail/wagtail/pull/4059/files#diff-984f91690294c77757dbe73c72481f9b), waiting for review.

**Comment by kaedroho** [comment]
That PR is huge and this is only a single line fix. Could we extract that fix into a separate PR and add a test?

**Comment by BertrandBordage** [comment]
@kaedroho: in fact, it’s already tested, but the postgres_search tests are disabled since a few weeks, when we started switching to Django 2.0. At the same time was merged my pull request re-implementing postgres_search without raw queries, introducing this bug that coincidentally never got tested on Travis. #4059 also restores testing on PostgreSQL by changing tox.ini.

Also, I agree that this PR is huge for just these fixes. I was just trying to put it forward so we can merge it soon and I can continue my work on adding autocomplete, row boosting, multi-language and so on :)
